### PR TITLE
Db solution

### PR DESCRIPTION
**Pull request by codecookerpro**
Moralis integration to the project

### CODE REVIEW

File: src/components/AccountDetails/index.tsx [hunk 1]
```diff
@@ -129,7 +130,7 @@ const AccountControl = styled.div`
   }
 `
 
-const AddressLink = styled(ExternalLink)<{ hasENS: boolean; isENS: boolean }>`
+const AddressLink = styled(ExternalLink) <{ hasENS: boolean; isENS: boolean }>`
```
Comment (0mkara) [inline src/components/AccountDetails/index.tsx:133]: `(ExternalLink) <{ hasENS:`
Type definityions should not have space between.

File: src/components/WalletModal/index.tsx [hunk 2]
```diff
@@ -134,11 +135,37 @@ export default function WalletModal({
   const openOptions = useCallback(() => {
     setWalletView(WALLET_VIEWS.OPTIONS)
   }, [setWalletView])
+  const { authenticate, isAuthenticated } = useMoralis();
 
+  const loginMoralis = async () => {
```
Comment (0mkara) [inline src/components/WalletModal/index.tsx:140]: Why do we need to login to moralis? I think this is the reason for the metamask moralis popup issue.

**Review by 0mkara — CHANGES_REQUESTED** [review]

File: src/pages/GoLivePage.tsx [hunk 3]
```diff
@@ -1,106 +1,222 @@
...
     dispatch(updatedeployedGovernorAddress(governorAddress))
-    navigate("/dashboard");
+    // navigate("/dashboard");
     setdeployedGovernor(governorAddress);
   }
+  const createToken = async () => {
+    const factory = await factoryCall(provider);
+    setisLoading(true);
+    onClose()
+    const creatingToken = await factory.createToken(tokenTitle, tokenSymbol, supply, holder, explain);
```
Comment (0mkara) [inline src/pages/GoLivePage.tsx:103]: When we create a transaction we should add the transaction to the account system.

**Review by 0mkara — CHANGES_REQUESTED** [review]

File: .env [hunk 4]
```diff
@@ -1,3 +1,7 @@
 REACT_APP_INFURA_KEY=97f712dc80bc4cfa82f75c7aceabb344
 REACT_APP_FORTMATIC_KEY=pk_test_A997503C0E4D3D18
-FACTORY_CONTRACT_ADDRESS=0xd940F6323F8287F49baE5dE14B6CbaCA641a48F5
\ No newline at end of file
+REACT_APP_FACTORY_CONTRACT_ADDRESS=0xd940F6323F8287F49baE5dE14B6CbaCA641a48F5
+REACT_APP_MORALIS_SERVER_URL="https://kwtfkw2l5s4j.usemoralis.com:2053/server"
+REACT_APP_MORALIS_APP_ID="NZe9KIusWvNeAN2U7D70bSFtDZRrBmI147OUoI79"
+REACT_APP_IPFS_ID=2DPQZ88vN6DworNjKmEULrEVRuo
+REACT_APP_IPFS_KEY=01493be23ac5ddd53beba8db5294d588
```
Comment (0mkara) [inline .env:7]: @codecookerpro pls do not commit variables. I will have to delete the app and recreate keys now. Our repo is public and we should not commit key variables to repositories.

File: src/utils/ipfs.ts [hunk 5]
```diff
@@ -23,26 +23,21 @@ export const fileUpload = async (coverImg: any) => {
...
-    fetchFile(cid);
-    resolve(cid);
+    const result = await (ipfs as IPFSHTTPClient).add(coverImg);
+    // fetchFile(cid);
+    resolve(result.path);
   });
 }
 
-export const fetchFile = async (cid: CID) => {
+export const fetchFile = async (cid: string) => {
```
Comment (0mkara) [inline src/utils/ipfs.ts:32]: cid is not `string`. Please use proper types. I would prefer
```
import { CID } from "ipfs-http-client";
cid: string | CID;
```

File: src/utils/ipfs.ts [hunk 6]
```diff
@@ -23,26 +23,21 @@ export const fileUpload = async (coverImg: any) => {
   return new Promise(async(resolve, reject)=> {
     console.log('Trying to upload file to IPFS');
     console.log(ipfs?.stats);
-    const { cid } = await (ipfs as IPFSHTTPClient).add(coverImg);
-    console.log(cid);
-    fetchFile(cid);
-    resolve(cid);
+    const result = await (ipfs as IPFSHTTPClient).add(coverImg);
```
Comment (0mkara) [inline src/utils/ipfs.ts:26]: Why should we typecast `ipfs as IPFSHTTPClient`? Instead check if `ipfs` is defined.

**Review by 0mkara — CHANGES_REQUESTED** [review]

File: .env (hunk 4 above)
Comment (codecookerpro) [reply .env:7]: i see.
i willnot commit those values in .env again.

**Review by 0mkara — CHANGES_REQUESTED** [review]

File: src/pages/GoLivePage.tsx [hunk 7]
```diff
@@ -1,105 +1,238 @@
...
+      console.log(error);
+    }
+  }
 
+  const DeployDAO = async (deployedTokenAddress: string) => {
+    console.log("DEploying Goerli")
+    console.log(deployedTokenAddress)
+    const creatingGovernor = factory && deployedTokenAddress !== null && await factory.createGovernor(title, deployedTokenAddress, title, purpose, longDesc, shortDesc);
+    const transactionInfo = await creatingGovernor.wait();
+    addTransaction(creatingGovernor, transactionInfo)
```
[the remark this answers is not in the document]
Comment (codecookerpro) [reply src/pages/GoLivePage.tsx:103]: const addTransaction = useTransactionAdder();
addTransaction(creatingGovernor, transactionInfo);
so it is the same as dispatch(addTransaction(creatingGovernor, transactionInfo) if you consider the useTransactionAdder() hook.

[the remark this answers is not in the document]
Comment (codecookerpro) [reply src/pages/GoLivePage.tsx:103]: const addTransaction = useTransactionAdder();
addTransaction(creatingGovernor, transactionInfo);
so it is the same as dispatch(addTransaction(creatingGovernor, transactionInfo) if you consider the useTransactionAdder() hook.

I am not clear what does it mean by "transaction needs to be added before transaction.wait".
Could you tell me about that?

**Pull request closed without merging**